src/app/modules/spcp/spcp.service.ts

```diff
@@ -218,6 +219,17 @@ export class SpcpService {
       if (isJwtPayload(payload, authType)) {
         return okAsync(payload)
       }
+      const payloadIsDefined = !!payload
+      const payloadKeys =
+        typeof payload === 'object' && !!payload && Object.keys(payload)
```

Comment:
nit: can reuse `payloadIsDefined`? (even though it has more letters lmao)

Reply:
unfortunately I had to do it this way because the types don't work out. if it's reused then the compiler complains that `payload` can be null haha

Reply:
isn't `!!payload` superfluous here, since objects are always truthy?

Reply:
nope!

<img width="491" alt="Screenshot 2020-12-11 at 11 36 25 AM" src="https://user-images.githubusercontent.com/29480346/101858705-1d2c0980-3ba5-11eb-9c50-5a5304369d97.png">